plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt

```diff
@@ -366,6 +375,26 @@ class CodeWhispererCodeScanSession(val sessionContext: CodeScanSessionContext) {
     private fun isAutoScan(): Boolean =
         sessionContext.codeAnalysisScope == CodeWhispererConstants.CodeAnalysisScope.FILE && !sessionContext.sessionConfig.isInitiatedByChat()
 
+    private fun generateScanName(): String {
+        val clientId = AwsSettings.getInstance().clientId
```

Comment:
Is scan name used anywhere other than metrics?

Reply:
Yes, in the CreateUploadUrl and CreateCodeScan APIs

Reply:
Hey, quick clarification: Why aren't we using the original UUIDs, what would be the difference here in using client ids? Since I see we are generating a new one at the end of this file?

Reply:
Using a UUID for every scan run means that the backend is unable to identify different runs of a scan belong to the same project. So here we want to introduce some way to identify that this scan is for a specific user, for a specific project, and for a specific scan type.

Reply:
the client id is already part of the user agent? why do we need to attach it again?

Reply:
@rli I guess technically we don't need to, but since `scanName` is exposed as an API field the client should generate the hash value